craft_application/services/provider.py

```diff
@@ -48,7 +48,7 @@
     from craft_application.services import ServiceFactory
 
 
-DEFAULT_FORWARD_ENVIRONMENT_VARIABLES: Iterable[str] = ()
+DEFAULT_FORWARD_ENVIRONMENT_VARIABLES: Iterable[str] = ("CRAFT_WORKSPACE_ID")
```

Comment:
I don't think this is the approach at all; shouldn't code that is interested in this `CRAFT_WORKSPACE_ID` add it to `DEFAULT_FORWARD_ENVIRONMENT_VARIABLES` itself? That would be similar to how the `register_snap()` classmethod is currently used